cocos/base/CCRefPtr.h

```diff
@@ -44,7 +44,7 @@ NS_CC_BEGIN
     {\
         if (ptr)\
         {\
-            const_cast<Ref*>(static_cast<const Ref*>(ptr))->retain();\
```

Comment:
The compiling error log:
```
/jenkins/workspace/ccs-pr/node/mac/build/../cocos/base/CCRefPtr.h:104:32: error: member function 'retain' not viable: 'this' argument has type 'const cocos2d::__String', but function is not marked const
        CC_REF_PTR_SAFE_RETAIN(_ptr);
                               ^~~~
/jenkins/workspace/ccs-pr/node/mac/build/../cocos/base/CCRefPtr.h:47:13: note: expanded from macro 'CC_REF_PTR_SAFE_RETAIN'
            ptr->retain();\
            ^
/jenkins/workspace/ccs-pr/node/mac/tests/cpp-tests/Classes/UnitTest/RefPtrTest.cpp:121:39: note: in instantiation of member function 'cocos2d::RefPtr<const cocos2d::__String>::RefPtr' requested here
        RefPtr<__String const> ref2 = __String::create("Hello");
                                      ^
```

So it seems should change to `const_cast<Ref*>(ptr)->retain()`